addonscript/src/addonscript/api.py

```diff
@@ -79,54 +128,61 @@ async def get_signed_addon_url(context, locale, pk):
         aiohttp.ClientError: If there is some sort of networking issue.
         asyncio.TimeoutError: If the network request times out.
 
-    Returns the signed XPI URL
+    Returns the XPI url, size and hash
 
     """
     # XXX Retry is done at top-level. Avoiding a retry here avoids never-ending retries cascade
-    upload_status = await get_upload_status(context, locale, pk)
-
-    if len(upload_status["files"]) != 1:
-        raise SignatureError("Expected 1 file. Got ({}) full response: {}".format(len(upload_status["files"]), upload_status))
-
-    if upload_status.get("validation_results"):
-        validation_errors = upload_status["validation_results"].get("errors")
-        if validation_errors:
-            raise FatalSignatureError("Automated validation produced errors: {}".format(validation_errors))
-
-    signed_data = upload_status["files"][0]
-
-    if not signed_data.get("signed"):
-        raise SignatureError('Expected XPI "signed" parameter. Got: {}'.format(signed_data))
+    version_detail = await get_version(context, locale, version_id)
+    file_detail = version_detail["file"]
+    status = file_detail["status"]
 
-    if not signed_data.get("download_url"):
-        raise SignatureError('Expected XPI "download_url" parameter. Got: {}'.format(signed_data))
+    if status == "disabled":
+        raise FatalSignatureError("XPI disabled on AMO")
+    if status != "public":
+        raise SignatureError("XPI not public")
+    return file_detail["url"], file_detail["size"], file_detail["hash"]
```

Comment:
I recommend using `namedtuple` for returning more a single value such as this, as it helps on readability and debugging.